test_regress/t/t_flag_lib_dpi.pl

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env perl
+if (!$::Driver) { use FindBin; exec("$FindBin::Bin/bootstrap.pl", @ARGV, $0); die; }
+# DESCRIPTION: Verilator: Verilog Test driver/expect definition
+#
+# This file ONLY is placed under the Creative Commons Public Domain, for
+# any use, without warranty, 2023 by Shupei Fan.
+# SPDX-License-Identifier: CC0-1.0
+
+scenarios(vlt => 1);
+
+run(logfile => "$Self->{obj_dir}/vlt_compile.log",
+    cmd => ["perl",
+            "$ENV{VERILATOR_ROOT}/bin/verilator",
+            "-cc",
+            "--build",
+            '--no-timing',
+            "-Mdir",
+            "$Self->{obj_dir}",
+            "t/t_flag_lib_dpi.v",
+            "$Self->{t_dir}/t_flag_lib_dpi.cpp",
+            "$Self->{t_dir}/t_flag_lib_dpi_main.cpp"],
+    verilator_run => 1,
+    );
+
+run(logfile => "$Self->{obj_dir}/cxx_compile.log",
+    cmd => ["cd $Self->{obj_dir}"
+            . " && $ENV{CXX} -o Vt_flag_lib_dpi libVt_flag_lib_dpi.a libverilated.a -lpthread -latomic"
```

Comment:
Instead of -lpthread -latomic which is not portable, use LIBS from the makefile.  There's no easy way to get this for a shell command, so I'd suggest make a little .mk makefile that has the appropriate rule and call it from here e.g. "make -C $Self->{obj_dir} $Self->{t_dir}/t_flag_lib_dpi.mk"